Jellyfin.Data/Entities/Security/ApiKey.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Globalization;
+
+namespace Jellyfin.Data.Entities.Security
+{
+    /// <summary>
+    /// An entity representing an API key.
+    /// </summary>
+    public class ApiKey
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ApiKey"/> class.
+        /// </summary>
+        /// <param name="name">The name.</param>
+        public ApiKey(string name)
+        {
+            Name = name;
+
+            AccessToken = Guid.NewGuid().ToString("N", CultureInfo.InvariantCulture);
+            DateCreated = DateTime.UtcNow;
+        }
+
+        /// <summary>
+        /// Gets the id.
+        /// </summary>
+        /// <remarks>
+        /// Identity, Indexed, Required.
+        /// </remarks>
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public int Id { get; private set; }
```

Comment:
This is a bit nitpick, but I prefer to set the access modifier to "protected".

EFCore requires them to be set as protected so they can be lazy-loaded, which is something we aren't doing now but who knows if in the future we want to.

Besides that, I found some issues in my side project where EFCore had some issues that got solved by switching from private to protected, and I wasn't using lazy loading. Not sure why, but I guess EFCore subclasses them internally and some complex objects or operations are messing with them. Either way, protected shouldn't hurt at all.

Reply:
Even if we were to use lazy loading, scalar properties are materialized by EF Core, not through lazy loading. I'm also unconvinced as to your subclassing argument, I don't see a problem with setting them as protected if that time comes. For now, I have this matching the rest of the entities in the model.